Vampire species for halloween

imtakingabreakdontatme opened the pull request:
:cl: Kor
add: Added vampires. They will be available as a roundstart race during the Halloween holiday event.
/:cl:

Vampires are another halloween species. 

They slowly lose blood constantly (unless in a coffin). They dust if they run out of blood.

They heal in coffins (but do not restore blood)

They can drain blood from other mobs they have in a passive grab.

They can turn into a bat.

They catch on fire if they enter the chapel.

Why: The more variety for halloween races the merrier. Having the background pressure of keeping your blood level up and finding "donors" without getting spaced by the AI might be fun as a limited time gimmick. This is also, bizarrely, one of the more commonly requested features as well.

imtakingabreakdontatme committed:
Vampires

imtakingabreakdontatme committed:
Action name

Dicks1 commented:
So I told Kor to add vampires for Halloween.... he actually did it the absolute madman
hahahahahahahahaha!

ChangelingRain commented on code/modules/mob/living/carbon/human/species_types/vampire.dm:
```diff
@@ -0,0 +1,84 @@
+			if(victim.stat == DEAD)
+				to_chat(H, "<span class='notice'>You need a living victim!</span>")
+				return
+			if(victim.blood_volume < 50)
```
this should be covered by the above check

KorPhaeron replied:
For whatever reason the check for instant blood volume death is only for humans

imtakingabreakdontatme committed:
Cooldown so you can't queue drains

imtakingabreakdontatme committed:
Blood drain rate

ChangelingRain commented on code/modules/mob/living/carbon/human/species_types/vampire.dm:
```diff
@@ -0,0 +1,90 @@
+	if(istype(C.loc, /obj/structure/closet/coffin))
+		C.heal_overall_damage(4,4)
+		C.adjustToxLoss(-4)
+		C.adjustOxyLoss(-4)
```
clone loss, too?

KorPhaeron replied:
This got added

ChangelingRain commented on code/modules/mob/living/carbon/human/species_types/vampire.dm:
```diff
@@ -0,0 +1,90 @@
+		C.adjustOxyLoss(-4)
+		return
+	C.blood_volume -= 1.5
+	if(C.blood_volume <= 0)
```
suffering from actual blood loss effects means this check won't trigger

ChangelingRain commented on code/modules/mob/living/carbon/human/species_types/vampire.dm:
```diff
@@ -0,0 +1,90 @@
+	invocation = "Squeak!"
+
+	shapeshift_type = /mob/living/simple_animal/hostile/retaliate/bat
+	list/current_shapes = list(/mob/living/simple_animal/hostile/retaliate/bat)
```
remove these `list/`s

JJRcop commented:
Make sure vampires can't be chaplains.

imtakingabreakdontatme committed:
Vampire cleanup/additional checks

Bawhoppen commented:
Where's the garlic, holy water, and wooden stakes weaknesses?

DoctoreMad commented:
>Garlic

Can we let our vampires not degrade into Friday the 13th tier?

Armhulen commented:
testmergeable?

Armhulen commented:
holy water should burn them, make humans rarely start as vampires (not an antag, just have to get blood to survive good luck!!

Grazyn commented:
I'm not convinved about them being roundstart, even for a limited 5-day event. How does this work rule-wise exactly? You literally have to harm people to survive, people who will obviously fight back or even kill you preventively, does this mean vampires are always valid? If you're a vampire and not antag, can you get banned for killing/incapacitating someone to feed yourself? If not, it means that being a vampire counts as being antag, and what's stopping every single player on the server from signing up as one?

JJRcop commented:
@Grayzn Doesn't make them an antag, and it doesn't make them valid. If you kill someone as a vamp, then that's excessive, since you can incapacitate them for a short time, or just find someone else if they put up too much of a fuss. If you are desperately running out of blood and you can't catch someone alive long enough to suck their blood, you just die, like plasmamen running out of plasma.

I don't think vamps get a free pass to kill by any means, you just need to drink blood to survive, and you can do that while still staying within the server's rules.

Grazyn commented:
Well I guess we'll see how it plays out, but keep in mind that currently any fight can easily escalate from throwing punches to someone ending up spaced. As non-antag vampire, you either find a willing volunteer or you must be prepared to die, because starting a fight with your victim means they are cleared to kill you while you can't do the same to them (after all, how can your victim know you're not gonna drain them dry once you have them stunlocked). I can't really picture an admin banning someone for killing a vampire who attacked them.

Grazyn commented:
Or let me put it this way: would you use lethal damage on someone who attacked you and cut off your leg? Do you think an admin would ban you for using lethal damage in that situation? Because the inconvenience you get from being low on blood is the same if not worse than having one leg cut off. You constantly pass out, you take oxy damage which will add to your preexisting damage, meaning you can easily end up in crit, you need to go to medbay to heal it and it will take some time, either with transfusion or iron.

Saying "You can kill a vamp if an IC fight has been properly escalated" is disingenuous, because the attempt itself will be enough to make the vamp valid.

Bawhoppen commented:
People could also not be dumb, that would probably solve it too

Grazyn commented:
The event will run for 5 days, this means hundreds of rounds, we can either address this issue now or deal with it then. "Hahahaha just not be dumb like nigga just let the vamp drain you haha" is not really addressing it.

JJRcop commented:
When I shapeshifted into a bat I couldn't change back.

pantherpusher commented:
What if people got greentext for "donating" blood to the vampire crew?

KorPhaeron commented:
@jjrcop

That would make sense since I gave the spell to the vampire mob

Whoops

Hatfish commented:
Keep up the masquerade or get dunked, it's not hard to understand neonates.

davethwave commented:
Also as a vampire is not a antag try to add some roundstart text saying so to prevent a load of self-antags starting on tg due to the confusion between this Halloween vampire and the antag one.

Armhulen commented:
rarely off of halloween humans start as vampires when??

JJRcop commented:
KorPhaeron#30

I gave vampires the ability to drink blood (the reagent) to restore their blood_volume

Ratvar commented:
Bloodtomato masterrace

DoctoreMad commented:
@Armhulen This feature is purely for fun, not to annoy people. I'd hate to be a vampire xenobiologist.

RemieRichards commented on code/modules/mob/living/carbon/human/species_types/vampire.dm:
```diff
@@ -0,0 +1,94 @@
+			if(victim.dna && ((NOBLOOD in victim.dna.species.species_traits) || victim.dna.species.exotic_bloodtype))
+				to_chat(H, "<span class='notice'>Your victim doesn't have blood!</span>")
+				return
+			if(victim.blood_volume < 50)
```
bit of an arbitrary limit.
could just say "you didn't get a full meal!" and pull all thats left.

KorPhaeron replied:
@JJRcop could you do this please? Simple but I will not have the chance today

KorPhaeron replied:
Also change the bat spell to try to grant to mind if you have the time

JJRcop replied:
Sure

KorPhaeron replied:
And maybe change V blood to U for Universal blood so it feels less snowflake

KorPhaeron replied:
Nevermind I found time to do this

Repukan commented:
Do they have the ability to stop time?

Armhulen commented:
@DoctoreMad get good, healing from coffins is pretty good

JJRcop committed:
Vampires can drink blood (reagent) (#30)

* New DRINKSBLOOD species_trait

All species can drink blood, but species with this trait
will restore lost blood_volume when drinking it

* Vampires have vampire blood and DRINKSBLOOD

People with vampire blood can take all other blood
Vampire blood can't be used for anyone that doesn't have it

KorPhaeron commented:
@armhulen

Healing from coffins (and turning into a bat) are also why I think this is too silly and off theme outside of halloween.

Durkel commented:
If the chaplains whip nullrod variant doesn't do bonus damage against vampires I will never have respect for you again.

DoctoreMad commented:
Demoniac race for Halloween when
Just a human that is possessed by a demon.

Repukan commented:
Make them take extra damage from holoparasites.

Bawhoppen commented:
>assistant breaks into medbay blood storage
>gets caught and tries to run
>"HE MUST BE A VAMP! GET HIM!"

sounds fun actually

davethwave commented:
remember to make them stronger than the average person too and perhaps have a minor upgrade relating to robustness or something like that, just make sure that at least give a upside or 2 to the vampire since for instance, by turning into a bat what help is it besides having a smaller hit box and perhaps being a bit faster.

davethwave commented:
Okay I retract my statement about the bat but still, a lot of downside not a lot of up.

KorPhaeron commented:
@davethwave free regen in coffins, transform spell that lets you ventcrawl, and night vision

Might already be too much for roundstart rather than too little honestly

davethwave commented:
wait, bats can ventcrawl?

KorPhaeron commented:
Yes

They can move in space as well

KorPhaeron commented:
Honestly I should probably remove night vision from them since they have no other interaction with light anyway

JJRcop committed:
Vampire tweaks (#31)

imtakingabreakdontatme committed:
Universal Blood

imtakingabreakdontatme committed:
Removes spell on species loss

davethwave commented:
then I retract all of my other statements about the vampire not being powerful, the power to vent crawl is already op if used properly.

imtakingabreakdontatme committed:
Vampire heart for checking blood level

JJRcop commented:
Dunno how you will feel about this but KorPhaeron#32

GunHog commented:
What if people that are succed dry by a vampire, then bitten again become vampires themselves?

KorPhaeron commented:
@grazyn

I plan to admin it like I would stealing a space suit in a crisis.

If you make a good faith effort to find monkeys/willing donors/sabutamol and you are still about to starve to death it will be okay to eat the clown

If you, roundstart, start eating the clown and kill him when he retaliates that wont be okay

I Am Legend scenarios caused by a traitor destroying medbay and the human heads of staff holed up in the chapel while the vampire crew riots for blood sounds fun

MoreRobustThanYou commented on code/modules/reagents/reagent_containers/blood_pack.dm:
```diff
@@ -65,6 +65,9 @@
 /obj/item/reagent_containers/blood/lizard
 	blood_type = "L"
 
+/obj/item/reagent_containers/blood/universal
```
> universal

WHY. JUST WHY?

davethwave commented:
And of course there is the mourge as well as cloning, even just asking genetics would probably be fine once the idea of vampire does not equal antag sets in throughout the crew.

KorPhaeron commented:
@tgstation/commit-access Halloween starts in about 24 hours on /tg/station so I am review/merge begging

JJRcop commented:
Vamps cån't suhh bluhh from deeed people, althhhough, you could rrrremooove the deeed person's bluhh ussssing an ivvvvyy drrrriiiip

FantasticFwoosh commented:
This seems like a level of detail that is unnecessary on a starting race for Halloween and more of a side-antagonist/experimental project @KorPhaeron 

Couldn't you tone them back a tiny bit with blood attached to nutrition or a blood subgroup of likable tastes (**including drinking at the bar bloody mary's / salbutomol / tomato soup etc**) and no explicit powers/particularly degrading weaknesses unless they are actually a antagonist?

- Holy water/chaplain weapon weakness, nonconvertible to cult for said reason but otherwise slightly accelerated  nutritional regeneration & a permanent eternal youth status? (as soon as that virus effect actually means anything)

There is a vampire mask floating around on other servers thematically to use if we just wanted to input a vampire antagonist status onto someone.

- Sucking blood out of someone is constituted as a harmful and (valid?) act without consent i guess.
